src/ui/map.js

```diff
@@ -1043,20 +1047,43 @@ class Map extends Camera {
      * @see [Example: Display a web map using an alternate projection](https://docs.mapbox.com/mapbox-gl-js/example/map-projection/)
      * @see [Example: Use different map projections for web maps](https://docs.mapbox.com/mapbox-gl-js/example/projections/)
      */
-    setProjection(projection?: ?ProjectionSpecification | string) {
+    setProjection(projection: ProjectionSpecification | string = {name: "mercator"}) {
```

Comment:
Are we at risk of breaking the public API with that particular function signature change? The only requirement here is that we should still allow `map.setProjection(null | undefined)` as mentioned in the docs as a valid call.

Reply:
Good catch Karim! It works on undefined but no longer on `map.setProjection(null)`. I'll fix this and add a test.

Reply:
Thanks! Yes any unit tests coverage increase of APIs touching projections is a good call, the coverage is still fairly low so we need to be careful.